.github/workflows/image.yaml

```diff
@@ -0,0 +1,128 @@
+# Build and push images to :
+# GitHub Container Registry (ghcr.io)
+# Red Hat Container Registry (quay.io)
+name: "Build Docker Images"
+
+on:
+  create: # Branch (so that the Docker image tags exist for a new branch when pushed)
+          # even if nothing on 'paths' below is changed.
+    branches:
+      - '**'
+
+  push:
+    paths:
+      - "**"
+      - ".github/workflows/image.yaml"
+
+    branches: # Only on paths above
+      - '**'
+
+    tags: # Always, regardless of paths above
+      - '*'
```

Comment:
more thoughts: we only want to build and push on creating + publishing a new tag right?

Reply:
I could see a scenario where you might want to build and push images that we are testing or experimenting with so something like this might work:
```yaml
on:
  push:
    branches:
      - main
    paths:
      - "Dockerfile.*"
      - "*/environment.yaml"
      - ".github/workflows/image.yaml"
    tags:
      - '*'
```

👆 this is based one of the first comments in this review